src/DefaultBuilder/src/WebApplicationBuilder.cs

```diff
@@ -416,4 +416,43 @@ private void ConfigureApplication(WebHostBuilderContext context, IApplicationBui
 
     void IHostApplicationBuilder.ConfigureContainer<TContainerBuilder>(IServiceProviderFactory<TContainerBuilder> factory, Action<TContainerBuilder>? configure) =>
         _hostApplicationBuilder.ConfigureContainer(factory, configure);
+
+    private void MergeMiddlewareDescriptions(IApplicationBuilder app)
```

Comment:
What's the startup overhead of this method?

Reply:
It should be small.

But how about this: This is only used for debugging. Wrap all the middleware description logic in `Debugger.IsAttached`. If there is no debugger attached then there is no overhead.

Reply:
Done.